### ISSUE TITLE
Error building Singular.jl on aarch64

### ISSUE THREAD

**Issue by rbehrends** [oscar-system/Singular.jl#272]
Building Singular on an ARM machine (aarch64) fails with the following error:

```
┌ Error: Error building `Singular`: 
│ [ Info: Downloading https://github.com/JuliaPackaging/Yggdrasil/releases/download/GMP-v6.1.2-1/GMP.v6.1.2.aarch64-linux-gnu.tar.gz to /tmp/devjulia/packages/Singular/5YqNQ/deps/usr/downloads/GMP.v6.1.2.aarch64-linux-gnu.tar.gz...
│ [ Info: Downloading https://github.com/JuliaPackaging/Yggdrasil/releases/download/MPFR-v4.0.2-1/MPFR.v4.0.2.aarch64-linux-gnu.tar.gz to /tmp/devjulia/packages/Singular/5YqNQ/deps/usr/downloads/MPFR.v4.0.2.aarch64-linux-gnu.tar.gz...
│ ERROR: LoadError: LoadError: Your platform ("aarch64-unknown-linux-gnu", parsed as "aarch64-linux-gnu-gcc7-cxx11") is not supported by this package!
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] top-level scope at /tmp/devjulia/packages/Singular/5YqNQ/deps/build_libflint.v0.0.0-ba0ceed35136a2a43441ab9a9b2e7764e38548ea.jl:27
│  [3] include(::Function, ::Module, ::String) at ./Base.jl:380
│  [4] include at ./Base.jl:368 [inlined]
│  [5] include(::String) at /tmp/devjulia/packages/Singular/5YqNQ/deps/build.jl:45
│  [6] top-level scope at /tmp/devjulia/packages/Singular/5YqNQ/deps/build.jl:45
│  [7] eval(::Module, ::Any) at ./boot.jl:331
│  [8] top-level scope at /tmp/devjulia/packages/Singular/5YqNQ/deps/build.jl:45
│  [9] include(::String) at ./client.jl:457
│  [10] top-level scope at none:5
│ in expression starting at /tmp/devjulia/packages/Singular/5YqNQ/deps/build_libflint.v0.0.0-ba0ceed35136a2a43441ab9a9b2e7764e38548ea.jl:23
│ in expression starting at /tmp/devjulia/packages/Singular/5YqNQ/deps/build.jl:19
└ @ Pkg.Operations /buildworker/worker/package_linuxaarch64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:949
```

The error seems to occur because there is no binary download for libflint for that architecture. I'm not sure if flagging it on Singular.jl is correct, but it seems to occur specifically in this context.

**Comment by thofma** [comment]
Which (ancient) version of Singular.jl is this? Surely not the most recent one.

**Comment by rbehrends** [comment]
It's what you get when trying to use `Pkg.add("Oscar")`. That's v0.3.2, unfortunately, but it's also what end users are seeing currently when installing Oscar.

**Comment by mohamed-barakat** [comment]
As I understand from Reimer this is also the reason why `HomalgProject` fails its tests on [oscar-ci-status-stable](https://oscar-system.github.io/oscar-ci-status-stable/) as its forced to use this old version of `Singular` dictated by the current released version of `Oscar`.

**Comment by rbehrends** [comment]
@mohamed-barakat Yeah, we had a separate issue with the GAP Digraphs package, but that was due to something server-related (and fixed in the meanwhile). I'm not positive that it's necessarily Singular.jl old that's a problem for HomalgProject.jl, but there seems to be a version issue with doctests, as the doctest errors disappear in the dev version. Generally, some of the package versions of dependent packages forced by `Pkg.add("Oscar")` are somewhat dated.

See [this log](https://oscarci.mathematik.uni-kl.de/job/oscar-stable/870/artifact/logs/build-870/CheckPackages.log) for what actual versions are in use for the stable version of Oscar.

**Comment by rbehrends** [comment]
There is also an error with v0.4.1, though that's a different one:

```
┌ Error: Error building `Singular`: 
│ ERROR: LoadError: LoadError: BoundsError: attempt to access 1-element Array{SubString{String},1} at index [3]
│ Stacktrace:
│  [1] getindex(::Array{SubString{String},1}, ::Int64) at ./array.jl:809
│  [2] (::var"#6#9")(::IOStream) at /tmp/devjulia/packages/Singular/cuKCT/deps/parselibs.jl:50
│  [3] open(::var"#6#9", ::String, ::Vararg{String,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./io.jl:325
│  [4] open(::Function, ::String, ::String) at ./io.jl:323
│  [5] (::var"#5#8")() at /tmp/devjulia/packages/Singular/cuKCT/deps/parselibs.jl:38
│  [6] cd(::var"#5#8", ::String) at ./file.jl:104
│  [7] top-level scope at /tmp/devjulia/packages/Singular/cuKCT/deps/parselibs.jl:37
│  [8] include(::String) at ./client.jl:457
│  [9] top-level scope at /tmp/devjulia/packages/Singular/cuKCT/deps/build.jl:52
│  [10] include(::String) at ./client.jl:457
│  [11] top-level scope at none:5
│ in expression starting at /tmp/devjulia/packages/Singular/cuKCT/deps/parselibs.jl:37
│ in expression starting at /tmp/devjulia/packages/Singular/cuKCT/deps/build.jl:52
└ @ Pkg.Operations /buildworker/worker/package_linuxaarch64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Operations.jl:949
```

**Comment by thofma** [comment]
Same architecture? Can you update the OP with julia version and Pkg.status?

**Comment by thofma** [comment]
PS: The fact that OSCAR.jl is using an old version of Singular.jl is not a Singular.jl issue.

**Comment by fingolfin** [comment]
@fieker we really really need to make an OSCAR.jl release.

**Comment by hannes14** [comment]
The bug with the String operation in parselib.jl (https://github.com/oscar-system/Singular.jl/issues/272#issuecomment-697304480)
was caused by a bug in libparse, fixed with https://github.com/Singular/Singular/commit/e6e50337e6f4b660421f7bbe12224078ee14afca

**Comment by fingolfin** [comment]
Updating Singular.jl with the fix in Singular is currently stalled (see <https://github.com/JuliaPackaging/Yggdrasil/pull/1718>). I hope that it can be unblocked soon, though (once Yggdrasil is update to a more recent BinaryBuilder development version.

**Comment by fingolfin** [comment]
This should be fixed in the latest Singular_jll (but beware, that one still has an issue on macOS; see https://github.com/JuliaPackaging/Yggdrasil/pull/1748 for a fix)

**Comment by fingolfin** [comment]
@rbehrends @mohamed-barakat can you please check if this is still an issue? It shouldn't be; it works for me on momo64.mathematik.uni-kl.de

It does *not* work on ARM32 right now, but that's because various binaries (including for Julia and CxxWrap...) are missing there.

**Comment by mohamed-barakat** [comment]
I am not aware of this being an issue anymore.
